File: org.kevoree.modeling.genetic.framework/src/main/java/org/kevoree/modeling/genetic/api/MutationOperator.java
package org.kevoree.modeling.genetic.api;


import org.kevoree.modeling.api.KMFContainer;

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 12/02/13
 * Time: 15:26
 */
public interface MutationOperator<A extends KMFContainer> {

    public void mutate(A parent);

}

File: org.kevoree.modeling.optimization.framework.core/src/main/java/org/kevoree/modeling/optimization/api/MutationOperator.kt
package org.kevoree.modeling.optimization.api;


import org.kevoree.modeling.api.KMFContainer;

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 12/02/13
 * Time: 15:26
 */
public trait MutationOperator<A : KMFContainer> {

    public fun mutate(parent : A);

}
